vm/systemSmartContracts/delegation.go

```diff
@@ -848,6 +851,25 @@ func (d *delegation) changeServiceFee(args *vmcommon.ContractCallInput) vmcommon
 		return vmcommon.UserError
 	}
 
+	if d.enableEpochsHandler.IsFlagEnabled(common.DelegationImprovementsV3Flag) {
+		currentEpoch := uint64(d.eei.BlockChainHook().CurrentEpoch())
```

Comment:
maybe you can extract the new code in a new function

Reply:
the function as a whole is simple enough I think, I would keep business logic inside.